controllers/azurecluster_reconciler.go

```diff
@@ -300,7 +300,10 @@ func (r *azureClusterReconciler) deleteNSG(ctx context.Context) error {
 
 // CreateOrUpdateNetworkAPIServerIP creates or updates public ip name and dns name
 func (r *azureClusterReconciler) createOrUpdateNetworkAPIServerIP() error {
```

Comment:
I think I missed this as part of the public IP refactor, it should probably move to PublicIPSpecs() and/or azure cluster controller to after the IP has been created since this should really be a status update (and we treat it as one in the controller)...